proposed/http-message.md

```diff
@@ -332,9 +505,11 @@ interface IncomingRequestInterface extends MessageInterface
      * purposes of how duplicate query parameters are handled, and how nested
      * sets are handled.
      *
-     * @return array
+     * @param array $query Array of query string arguments, typically from
+     *     $_GET.
+     * @return void
      */
-    public function getQueryParams();
+    public function setQueryParams(array $query);
```

Comment:
I think in real use-cases this is a bit problematic though.

Think about this: what if this is used in the context of a client. I set setQueryParams to pass a few variables. Would it alter the URI that the client ends up sending?

Another example, a middleware wants know whether a certain query parameter was set. Should it be forced to check both the uri and the query parameters?